services/scanners/task_parser.go

```diff
@@ -414,37 +437,56 @@ func (p *ParserTask) getAccountBalance(height int64, address stakingAPI.Address)
 		return balance, err
 	}
 
-	delegations, err := p.stakingAPI.DelegationsFor(p.ctx, &stakingAPI.OwnerQuery{
+	delegations, err := p.stakingAPI.DelegationInfosFor(p.ctx, &stakingAPI.OwnerQuery{
 		Height: height,
 		Owner:  address,
 	})
 
 	var delegationsBalance uint64
 	for _, balance := range delegations {
-		delegationsBalance += balance.Shares.ToBigInt().Uint64()
+
+		stakeBalance, err := balance.Pool.StakeForShares(&balance.Shares)
+		if err != nil {
+			log.Error("Somehow delegations rpc values is wrong", zap.Error(err))
+			continue
+		}
+
+		delegationsBalance += stakeBalance.ToBigInt().Uint64()
 	}
 
-	debondingDelegations, err := p.stakingAPI.DebondingDelegationsFor(p.ctx, &stakingAPI.OwnerQuery{
+	debondingDelegations, err := p.stakingAPI.DebondingDelegationInfosFor(p.ctx, &stakingAPI.OwnerQuery{
 		Height: height,
 		Owner:  address,
 	})
 
 	var debondingDelegationsBalance uint64
 	for _, debondings := range debondingDelegations {
 		for _, value := range debondings {
-			debondingDelegationsBalance += value.Shares.ToBigInt().Uint64()
+
+			debondingBalance, err := value.Pool.StakeForShares(&value.Shares)
+			if err != nil {
+				log.Error("Somehow debonding rpc values is wrong", zap.Error(err))
+				continue
+			}
+
+			debondingDelegationsBalance += debondingBalance.ToBigInt().Uint64()
 		}
 	}
 
+	//TODO handle selfstake
```

Comment:
Is the plan to break up `EscrowBalanceActive`/`EscrowBalanceShare` (and similarly `EscrowDebondingActive`/`EscrowDebondingShare`) to show incoming delegations from others and self-delegation separately?

I would suggest splitting it up into something like:
- `IncomingDelegationsBalance` / `IncomingDelegationsShares`
- `SelfDelegationBalance` / `SelfDelegationShares`

And similarly for incomding debonding delegations vs. self-debonding delegations.